Implementation for SBOL2<-->SBOL3 converter

Yehuda-Binik opened the pull request:
In the new converter, I completed the `visit_implementation` function for both the `SBOL3To2ConversionVisitor` and `SBOL2To3ConversionVisitor` classes. I also added two test cases to test my code using the SBOL3 [implementation.nt](https://github.com/SynBioDex/SBOLTestSuite/blob/master/SBOL3/entity/implementation/implementation.nt) file from the [SBOLTestSuite](https://github.com/SynBioDex/SBOLTestSuite/tree/master).

Yehuda-Binik committed:
Worked on implementation 3 -> 2 for the updated version of the converter

Yehuda-Binik committed:
Merge remote-tracking branch 'origin/develop' into implementation3to2

Yehuda-Binik committed:
Working on solving a bug in code

Yehuda-Binik committed:
Please enter the commit message for your changes. Lines starting

Yehuda-Binik committed:
Fixed 'built' link bug

Yehuda-Binik committed:
Fixed backport version bug and brought back original test cases

Yehuda-Binik committed:
Deleted unused test files

Yehuda-Binik committed:
changed the formatting of variable names in visit_implementation() in the SBOL3To2ConversionVisitor class to comply with variable naming conventions

jakebeal commented:
@Yehuda-Binik : there are some files that shouldn't have been checked in that were checked in. Can you please remove the `.idea` and `sbol_utilities.egg-info` directories?  In general, `.idea` and `*.egg-info` should be in your `.gitignore` files.

Yehuda-Binik committed:
Update .gitignore to ignore .idea and *.egg-info directories

Yehuda-Binik committed:
removed .idea and sbol_utilities.egg-info directories

Yehuda-Binik commented:
@jakebeal I removed the `.idea` and `sbol_utilities.egg-info` directories. I also updated my `.gitignore` Is it necessary to delete my `.gitignore` file from the pull request as well? Let me know if there is anything left I need to change.

jakebeal commented:
Thank you! Adding to .gitignore is fine and helpful. I will review the PR in detail now.

jakebeal commented on sbol_utilities/sbol3_sbol2_conversion.py:
```diff
@@ -462,9 +467,15 @@ def visit_generic_location(self, a: sbol2.GenericLocation):
+        # Map over all other TopLevel properties and extensions not covered by the constructor
+        self._convert_toplevel(imp2, imp3)
+
+        # raise NotImplementedError('Conversion of Implementation from SBOL2 to SBOL3 not yet implemented')
```
Delete this comment as unneeded.

jakebeal commented on test/sbol3.nt:
This file is out of place: it's in the test directory, not the test_files subdirectory. It looks like an accidental check-in?

jakebeal commented on test/test_files/sbol3_genbank_conversion/ignoring_sbol_properties.nt.test:
This file also appears to be an accidental check-in.

jakebeal commented on test/test_sbol2_sbol3_direct.py:
```diff
@@ -11,15 +11,15 @@
 
 TEST_FILES = Path(__file__).parent / 'test_files'
 
-
```
Please avoid making un-needed changes in spacing: it increases the likelihood of conflicts.
There are several locations, here and below.

jakebeal commented on test/test_sbol2_sbol3_direct.py:
```diff
@@ -35,8 +35,8 @@ def test_3to2_conversion(self):
     def test_2to3_conversion(self):
-        """Test ability to convert a simple part from SBOL3 to SBOL2"""
-        # Load an SBOL3 document and check its contents
+        """Test ability to convert a simple part from SBOL2 to SBOL3"""
```
Good catch!

jakebeal reviewed (changes requested):
There's some minor cleanup needed - the substance of the changes is all good, I just need you to remove the accidental check-ins and unnecessary space changes.

Yehuda-Binik committed:
Update sbol3_sbol2_conversion.py

Removed unnecessary comment

Yehuda-Binik committed:
Delete test/sbol3.nt

Deleted a file that was accidentally checked in

Yehuda-Binik committed:
Delete test/test_files/sbol3_genbank_conversion/ignoring_sbol_properties.nt.test

This file was accidentally checked in.

Yehuda-Binik committed:
Update test_sbol2_sbol3_direct.py

Removed unnecessary and superfluous formatting changes.

Yehuda-Binik committed:
Update test_sbol2_sbol3_direct.py

Removed two commented out lines of code that were useful during testing, but no longer needed

Yehuda-Binik commented:
I removed the two accidentally checked-in files deleted 3 unnecessary comments (the unneeded `raise NotImplemented` comment as well as two comments I had written myself) and removed superfluous formatting changes.

jakebeal reviewed (approved):
Code looks good, and the two errors are, I believe, only from the secret-requiring tests that are not affected by this PR
